### ISSUE TITLE
CH$LINK and multiple identifiers

### ISSUE THREAD

**Issue by schymane** [MassBank/MassBank-web#80]
We will end up with some cases where we will have multiple valid DTXSIDs for MassBank spectra, especially for the Literature Specs and Eawag Additional. DTXSID for individual (representative) structure, then for (sub)groups and related classes. How should we deal with this in the record? The Record Format only has repeated identifiers for PubChem but there we have either CID or SID to precede the identifiers.
Like so:
CH$LINK: COMPTOX DTXSID4059091 DTXSID0024842 
Or so:
CH$LINK: COMPTOX DTXSID4059091 
CH$LINK: COMPTOX_CLASS DTXSID0024842 
Personally I'd prefer the former, if we can dispatch/create links/search correctly like that. The class associations are likely to get quite complicated (several classes/mixtures associated with one structure, on several levels).

For this example:
https://comptox.epa.gov/dashboard/dsstoxdb/results?utf8=%E2%9C%93&search=DTXSID4059091 is one structure that belongs under this group:
https://comptox.epa.gov/dashboard/dsstoxdb/results?utf8=%E2%9C%93&search=COCONUT+FATTY+ACID+DIETHANOLAMIDE

**Comment by meowcat** [comment]
I personally would prefer:
CH$LINK: COMPTOX DTXSID4059091
CH$LINK: COMPTOX DTXSID0024842

i.e. just multiple COMPTOX entries, rather than one with multiple IDs.

**Comment by Treutler** [comment]
I agree to @meowcat to expand multiple IDs to multiple entries. Hence, I see no problems with the database and the MassBank website. Can we close this issue?

**Comment by Treutler** [comment]
The multiple link format is not covered by the record specification and implemented accordingly in MassBank-web. Multiple links need to be specified one by one.

**Comment by ChemConnector** [comment]
Just wondering whether there was an issue regarding adding DTXSID outlinks that you are waiting on the EPA to resolve? I am assuming that you may be waiting on an InChIKey to DTXSID service but am not sure how you are forming the links?

**Comment by schymane** [comment]
Links are done web side, but we need DTXSID fields before we set up the links. So InChIKey to DTXSID web service can help us fix that for all records with matching entire, then making the links is very easy ...

-------------------------------------------
PI: EnvCheminf @ LCSB
FNR ATTRACT Fellow
emma.schymanski@uni.lu




On Fri, Apr 26, 2019 at 11:10 PM +0200, "ChemConnector" <notifications@github.com<mailto:notifications@github.com>> wrote:


Just wondering whether there was an issue regarding adding DTXSID outlinks that you are waiting on the EPA to resolve? I am assuming that you may be waiting on an InChIKey to DTXSID service but am not sure how you are forming the links?

**Comment by schymane** [comment]
Issue re web service here to get this done data-side:
https://github.com/MassBank/MassBank-data/issues/66

One outstanding issue before we can close this ... @tsufz are you able to make sure that the DTXSIDs will have active hyperlinks when they are in the records? I can't recall if we've already done this ...

Pattern is easy:
https://comptox.epa.gov/dashboard/dsstoxdb/results?search=DTXSID5039224

**Comment by Treutler** [comment]
The hyperlink is already included:
https://github.com/MassBank/MassBank-web/blob/9fd2ddf5bd3240d3cab8dca1e1f228c044008728/MassBank-Project/MassBank/src/main/webapp/RecordDisplay.jsp#L350

**Comment by schymane** [comment]
Great, thanks for confirming ... I couldn't find a record with DTXSID easily to check ;-)
Closing now ...
